### PR TITLE
chore: removing error state for rejecting chain switch

### PR DESCRIPTION
**Pull request by JackShort**
## Description
* removed error alert from web3 status
* if user rejected switch do not throw error/toast

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2211/updates-to-network-switch-error

## Screen capture

https://github.com/Uniswap/interface/assets/9094792/d30fc674-e334-4b2e-a20c-c2db58778f42

### CODE REVIEW

File: src/components/Web3Status/index.tsx [hunk 1]
```diff
@@ -174,15 +153,6 @@ function Web3StatusInner() {
...
     return null
-  } else if (error) {
-    return (
-      <Web3StatusError onClick={handleWalletDropdownClick}>
-        <NetworkIcon />
-        <Text>
-          <Trans>Error</Trans>
-        </Text>
-      </Web3StatusError>
-    )
```
Comment (JFrankfurt) [inline src/components/Web3Status/index.tsx:185]: Is there any more generic error (not user rejection) that we do want to preserve this error for? (I'm thinking like the wallet genuinely failed to switch...?)

**Review by JFrankfurt — APPROVED** [review]

File: src/components/Web3Status/index.tsx (hunk 1 above)
Comment (JackShort) [reply src/components/Web3Status/index.tsx:185]: yeah in the linear ticket it actually said to remove that for this and just show a toast

File: src/hooks/useSelectChain.ts [hunk 2]
```diff
@@ -16,12 +17,17 @@ export default function useSelectChain() {
...
         dispatch(updateConnectionError({ connectionType, error: undefined }))
         await switchChain(connector, targetChain)
       } catch (error) {
+        if (didUserReject(connection, error)) {
+          return
+        }
+
         console.error('Failed to switch networks', error)
 
         dispatch(updateConnectionError({ connectionType, error: error.message }))
```
Comment (zzmp) [inline src/hooks/useSelectChain.ts:33]: What happens with this connection error? Is it still used?

File: src/hooks/useSelectChain.ts [hunk 3]
```diff
@@ -16,12 +17,17 @@ export default function useSelectChain() {
...
 
-      const connectionType = getConnection(connector).type
+      const connection = getConnection(connector)
+      const connectionType = connection.type
 
       try {
         dispatch(updateConnectionError({ connectionType, error: undefined }))
         await switchChain(connector, targetChain)
       } catch (error) {
+        if (didUserReject(connection, error)) {
```
Comment (zzmp) [inline src/hooks/useSelectChain.ts:27]: If the user did not reject, but the chain change failed, shouldn't we still show the error state?

**Review by zzmp — CHANGES_REQUESTED** [review]
FYI I have a [pending PR](https://github.com/Uniswap/interface/pull/6662) to add these flows to integration tests. It is pending updates to the underlying test framework (cypress-hardhat), but I can update those tests with whatever changes you make here :)

File: src/hooks/useSelectChain.ts (hunk 2 above)
Comment (JackShort) [reply src/hooks/useSelectChain.ts:33]: Yeah it still goes through if there is a connection error the toast still should pop up

File: src/hooks/useSelectChain.ts (hunk 3 above)
Comment (JackShort) [reply src/hooks/useSelectChain.ts:27]: A toast should pop up which is what happens here

File: src/hooks/useSelectChain.ts (hunk 2 above)
Comment (JackShort) [reply src/hooks/useSelectChain.ts:33]: This gets picked up by our sentry logger that's the reason I kept it around

Comment (zzmp) [reply src/hooks/useSelectChain.ts:33]: If this is only here for Sentry, can you just explicitly call Sentry to report the exception? No need to go the roundabout way and add complexity.

**Review by zzmp — CHANGES_REQUESTED** [review]
Could you report the sentry error directly, and remove the unused connection error state? Or is there some reason it should still be used (like other infra that depends on it)?

**Comment by JackShort** [comment]
> Could you report the sentry error directly, and remove the unused connection error state? Or is there some reason it should still be used (like other infra that depends on it)?

Actually it was being store as part of the state in sentry -  so I removed it since we literally don't use it anywhere now and its logging an error thats being picked up so I don't really see any point in it being part of the sentry state

**Review by zzmp — APPROVED** [review]

**Pull request merged**